crates/radix-engine-toolkit-uniffi/src/transaction/message.rs

```diff
@@ -61,7 +61,41 @@ pub enum DecryptorsByCurve {
 }
 
 pub type AesWrapped128BitKey = Vec<u8>;
-pub type PublicKeyFingerprint = Vec<u8>;
+
+#[derive(Clone, Debug, Hash, Eq, PartialEq)]
+pub struct PublicKeyFingerprint(String);
```

Comment:
Why use `String` instead of `Vec<u8>` - it is bytes, so changing it to `String` feels like a degradation?

Reply:
`Vec<u8>` is converted to byte slice in Go (`[]byte`) which is non-comparable and cannot be used as hash map key. So I had to change that type to something else.

Reply:
@mstrug-rdx sorry I was reviewing from my personal lap top on my lunch earlier.... Right, so `Go` has the same challenge as `Kotlin` - I solved it like this in Sargon:
https://github.com/radixdlt/sargon/blob/main/src/core/types/bag_of_bytes.rs#L39-L58

Which I then in `uniffi.toml` make into a `Data` in Swift again, meaning the time has zero impact in Swift - and other languages where it is not a problem:
https://github.com/radixdlt/sargon/blob/main/uniffi.toml#L7-L17
(but now `equals` works in `Kotlin` (and in `Go`)).

What is nice about this is that I can continue to work with `Vec<u8>` in Rust, Swift is unaffected and now also works in Kotlin.

Understood if you dont wanna introduce a corresponding `BagOfBytes` in RET, but I wanted to share that with you, showing that we had that challenge in Sargon too but overcame it with that type.

Reply:
Thanks for the examples for Kotlin and Swift, I think I will be able to do similar conversion for Go. We will have `Vec<u8>` in Rust and `string` in Go as `HashMap` key.